src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,913 @@
-1<2+ {_} {[]}     = 1<[23]
-1<2+ {_} {b ∷ bs} = <-trans 1<[23] (∷ʳ-mono-< {a = b} 1<2+)
-
-0<1+ : ∀ {bs} → 0# < bs 1#
-0<1+ {[]}     = less (s≤s z≤n)
-0<1+ {b ∷ bs} = <-trans (less (s≤s z≤n)) 1<2+
-
-<⇒≢ : ∀ {a b} → a < b → a ≢ b
-<⇒≢ lt eq = asym⟶irr (resp₂ _<_) sym <-asym eq lt
-
-<-cmp : Trichotomous _≡_ _<_
-<-cmp 0#            0#            = tri≈ (<-irrefl refl) refl (<-irrefl refl)
-<-cmp 0#            (_ 1#)        = tri< 0<1+ (<⇒≢ 0<1+) (<-asym 0<1+)
-<-cmp (_ 1#)        0#            = tri> (<-asym 0<1+) (<⇒≢ 0<1+ ∘ sym) 0<1+
-<-cmp ([] 1#)       ([] 1#)       = tri≈ (<-irrefl refl) refl (<-irrefl refl)
-<-cmp ([] 1#)       ((b ∷ bs) 1#) = tri< 1<2+ (<⇒≢ 1<2+) (<-asym 1<2+)
-<-cmp ((a ∷ as) 1#) ([] 1#)       = tri> (<-asym 1<2+) (<⇒≢ 1<2+ ∘ sym) 1<2+
-<-cmp ((a ∷ as) 1#) ((b ∷ bs) 1#) with <-cmp (as 1#) (bs 1#)
-... | tri<  lt ¬eq ¬gt =
-  tri< (∷ʳ-mono-< lt)  (<⇒≢ (∷ʳ-mono-< lt)) (<-asym (∷ʳ-mono-< lt))
-... | tri> ¬lt ¬eq  gt =
-  tri> (<-asym (∷ʳ-mono-< gt)) (<⇒≢ (∷ʳ-mono-< gt) ∘ sym) (∷ʳ-mono-< gt)
-... | tri≈ ¬lt refl ¬gt with 𝔽ₚ.<-cmp a b
-...   | tri≈ ¬lt′ refl ¬gt′ =
-  tri≈ (<-irrefl refl) refl (<-irrefl refl)
-...   | tri<  lt′ ¬eq  ¬gt′ =
-  tri< (∷ˡ-mono-< lt′)  (<⇒≢ (∷ˡ-mono-< lt′)) (<-asym (∷ˡ-mono-< lt′))
-...   | tri> ¬lt′ ¬eq  gt′  =
-  tri> (<-asym (∷ˡ-mono-< gt′)) (<⇒≢ (∷ˡ-mono-< gt′) ∘ sym) (∷ˡ-mono-< gt′)
-
-<-isStrictTotalOrder : IsStrictTotalOrder _≡_ _<_
-<-isStrictTotalOrder = record
++-isSemigroup : IsSemigroup _≡_ _+_
++-isSemigroup = record
+  { isMagma = +-isMagma
+  ; assoc = +-assoc
+  }
+
++-semigroup : Semigroup 0ℓ 0ℓ
++-semigroup = record
+  { isSemigroup = +-isSemigroup
+  }
+
++-0-isMonoid : IsMonoid _≡_ _+_ 0#
++-0-isMonoid = record
+  { isSemigroup = +-isSemigroup
+  ; identity    = +-identity
+  }
+
++-0-monoid : Monoid 0ℓ 0ℓ
++-0-monoid = record
+  { ε        = 0#
+  ; isMonoid = +-0-isMonoid
+  }
+
++-0-isCommutativeMonoid : IsCommutativeMonoid _≡_ _+_ 0#
++-0-isCommutativeMonoid = record
+  { isSemigroup = +-isSemigroup
+  ; identityˡ   = +-identityˡ
+  ; comm        = +-comm
+  }
+
++-0-commutativeMonoid : CommutativeMonoid 0ℓ 0ℓ
++-0-commutativeMonoid = record
+  { isCommutativeMonoid = +-0-isCommutativeMonoid
+  }
+
+
+
+------------------------------------------------------------------------------
+-- Properties of multiplication, toℕ, fromℕ. Chapter II.
+------------------------------------------------------------------------------
+
+private  2*ₙ2*ₙ =  (2 *ₙ_) ∘ (2 *ₙ_)
+
+
+toℕ-homo-* :  ∀ x y → toℕ (x * y) ≡ toℕ x *ₙ toℕ y
+toℕ-homo-* x y =  aux x y (size x +ₙ size y) ℕp.≤-refl
+  where
+  aux :  (x y : Bin) → (cnt : ℕ) → (size x +ₙ size y ≤ₙ cnt) →
+         toℕ (x * y) ≡ toℕ x *ₙ toℕ y
+
+  aux 0# _  _ _ = refl
+  aux x  0# _ _ = begin
+    toℕ (x * 0#)      ≡⟨ cong toℕ (*-zeroʳ x) ⟩
+    0                 ≡⟨ sym (ℕp.*-zeroʳ (toℕ x)) ⟩
+    toℕ x *ₙ toℕ 0#   ∎
+
+  ---------------------
+  aux (2suc x) (2suc y) (1+ cnt) (s≤s |x|+1+|y|≤cnt) = begin
```

Comment:
Please could you make sure the `≡⟨` are aligned in all the different cases in this proof.